src/texture_stage.cpp

```diff
@@ -228,6 +228,31 @@ int TextureStage::SetTexture(const SDL_Surface *surface, uint8_t *memory_base) c
           }
         }
       } break;
+      case NV097_SET_TEXTURE_FORMAT_COLOR_L_DXT1_A1R5G5B5: {
+        uint32_t *source = pixels;
+        for (int y = 0; y < surface->h; y += 4) {
+          for (int x = 0; x < surface->w; x += 4, source += 4) {
+            // TODO: use proper encoding of the colors by quering all the colors and calculating color distances
```

Comment:
I wonder if it'd be easier to bring in an existing implementation like https://github.com/Cyan4973/RygsDXTc as a submodule in third_party. Looks like that particular one can also do DXT5 which might allow easy expansion of this test.

Reply:
This one might take a little more time to do than the others, so i likely update the PR without bringing in the new lib, but I will look into it on the way.

Reply:
No problem, I'm also happy to merge with your partial implementation and add the third_party lib later.